test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ActionOverloadingTests/Tests/ActionOverloadingEndToEndTests.cs

```diff
@@ -0,0 +1,266 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="ActionOverloadingEndToEndTests.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System.Xml;
+using Microsoft.AspNetCore.OData;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData.Client.E2E.TestCommon;
+using Microsoft.OData.Client.E2E.Tests.ActionOverloadingTests.Server;
+using Microsoft.OData.Client.E2E.Tests.Common.Clients.EndToEnd.Default;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd;
+using Microsoft.OData.Edm;
+using Microsoft.OData.Edm.Csdl;
+using Microsoft.OData.Edm.Validation;
+using Xunit;
+
+namespace Microsoft.OData.Client.E2E.Tests.ActionOverloadingTests.Tests
+{
+    [TestCaseOrderer("Microsoft.OData.Client.E2E.TestCommon.PriorityOrderer", "Microsoft.OData.Client.E2E.TestCommon")]
+    public class ActionOverloadingEndToEndTests : EndToEndTestBase<ActionOverloadingEndToEndTests.TestsStartup>
+    {
+        private readonly Uri _baseUri;
+        private readonly Container _context;
+        public class TestsStartup : TestStartupBase
+        {
+            public override void ConfigureServices(IServiceCollection services)
+            {
+                services.ConfigureControllers(typeof(ProductsController), typeof(OrderLinesController), typeof(PeopleController));
+
+                services.AddControllers().AddOData(opt => opt.Count().Filter().Expand().Select().OrderBy().SetMaxTop(null)
+                    .AddRouteComponents("odata", CommonEndToEndEdmModel.GetEdmModel()));
+            }
+        }
+        public ActionOverloadingEndToEndTests(TestWebApplicationFactory<TestsStartup> fixture)
+            : base(fixture)
+        {
+            _baseUri = new Uri(Client.BaseAddress, "odata/");
+            _context = new Container(_baseUri);
+            _context.HttpClientFactory = HttpClientFactory;
+        }
+
+        [Fact]
+        public void ExecuteOverloadedOperations()
+        {
+            var product = _context.Execute<Common.Clients.EndToEnd.Product>(new Uri(_baseUri + "Products(-10)", UriKind.Absolute)).Single();
+            OperationDescriptor productOperationDescriptor = _context.GetEntityDescriptor(product).OperationDescriptors.Single();

```

Comment:
Maybe we should retrieve the descriptor by name instead of `Single()`, that way adding new operations to the model would not cause this test to fail. It would also make the intent clearer in my opinion.

Reply:
If it's critical to this test that there only be exactly one overload of that operation, then maybe use Single with a filter against the name instead of an unfiltered `Single`.